src/main.rs

```diff
@@ -1,32 +1,67 @@
+use std::io::{Stdin, Stdout, Write};
+
 fn main() {
+    let mut terminal = Terminal::new();
     loop {
-        if generator_todo() {
-            break;
-        }
+        terminal.ask_if_new_todo();
+        let new_todo = terminal.ask_for_new_todo();
+        terminal.show_todo(&new_todo);
     }
 }
 
-fn input() -> String {
-    let mut buf = String::new();
-    std::io::stdin().read_line(&mut buf).unwrap();
-    buf.trim().to_string()
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
 }
 
-fn generator_todo() -> bool {
-    println!("Você deseja criar um novo TODO? (insira apenas n ou s)");
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
 
-    let response = input();
+    fn input(&self) -> String {
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+        buf.trim().to_string()
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
```

Comment:
Aparentemente não tem necessidade disso ser `&mut self`
```suggestion
    fn ask_for_new_todo(&self) -> Todo {
```

Reply:
Feito